extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+					$this->addWarning( wfMessage( 'userrenametool-page-exists', $newPage->getText() )->inContentLanguage()->text() );
+				}
+				else {
+					$this->addLog( "Moving page " . $oldPage->getText() . " in namespace {$row->page_namespace} to " . $newTitle->getText() );
+					$success = $oldPage->moveTo( $newPage, false,  wfMessage( 'userrenametool-move-log', $oldTitle->getText(), $newTitle->getText() )->inContentLanguage()->text() );
+
+					if ( $success === true ) {
+						$this->addLog( "Updating wiki \"{$cityDb}\": User page " . $oldPage->getText() . " moved to " . $newPage->getText() . '.' );
+					} else {
+						$this->addLog( "Updating wiki \"{$cityDb}\": User page " . $oldPage->getText() . " could not be moved to " . $newPage->getText() . '.' );
+						$this->addWarning( wfMessage( 'userrenametool-page-unmoved', array( $oldPage->getText(), $newPage->getText() ) )->inContentLanguage()->text() );
+					}
+				}
+			}
+			$dbw->freeResult( $pages );
+		} catch ( DBError $e ) {
+			// re-throw DB related exceptions instead of silently ignoring them (@see PLATFORM-775)
+			throw $e;
+		} catch ( Exception $e ) {
+			$this->addLog( "Exception while moving pages: " . $e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() );
+		}
+		/* End of move user pages */
+
+		foreach ( $tasks as $task ) {
+			$this->addLog( "Updating wiki \"{$cityDb}\": {$task['table']}:{$task['username_column']}" );
+			$this->renameInTable( $dbw, $task['table'], $this->mUserId, $this->mOldUsername, $this->mNewUsername, $task );
+		}
+
+		/* Reset local editcount */
+		$this->resetEditCountWiki();
+
+		$hookName = 'UserRename::AfterLocal';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, array( $dbw, $this->mUserId, $this->mOldUsername, $this->mNewUsername, $this, $wgCityId, &$tasks ) );
+
+		$dbw->commit();
+
+		// Save entry in local Special:Log
+		// Important: assuming that run inside the maintenance script
+		$this->addLocalLog( wfMessage( 'userrenametool-success', $this->mOldUsername, $this->mNewUsername )->inContentLanguage()->text() );
+
+		$this->addLog( "Finished updating wiki database: {$cityDb}" );
+
+		$this->addMainLog( "log", RenameUserLogFormatter::wiki( $this->mRequestorName, $this->mOldUsername, $this->mNewUsername, $wgCityId, $this->mReason,
+			!empty( $this->warnings ) || !empty( $this->errors ) ) );
+
+		$this->addLog( "Invalidate user data on local Wiki ({$wgCityId}): {$this->mOldUsername}" );
+		$this->invalidateUser( $this->mOldUsername );
+
+		$this->addLog( "Invalidate user data on local Wiki ({$wgCityId}): {$this->mNewUsername}" );
+		$this->invalidateUser( $this->mNewUsername );
+
+		$wgUser = $wgOldUser;
+	}
+
+	/**
+	 * Processes specific local wiki database and makes all needed changes for an IP address
+	 *
+	 * Important: should only be run within maintenace script (bound to specified wiki)
+	 */
+	public function updateLocalIP() {
+		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app